src/main/java/de/tum/in/www1/artemis/web/rest/lecture/AttachmentUnitResource.java

```diff
@@ -161,30 +164,50 @@ public ResponseEntity<AttachmentUnit> createAttachmentUnit(@PathVariable Long le
     }
 
     /**
-     * POST lectures/:lectureId/attachment-units/split : creates new attachment units. The provided file must be a pdf file.
+     * POST lectures/:lectureId/process-units/upload : Temporarily uploads a file which will be processed into lecture units
      *
-     * @param lectureId                 the id of the lecture to which the attachment units should be added
-     * @param lectureUnitInformationDTO the units that should be created
-     * @param file                      the file to be splitted
-     * @return the ResponseEntity with status 200 (ok) and with body the newly created attachment units
+     * @param file      the file that will be processed
+     * @param lectureId the id of the lecture to which the attachment units will be added
+     * @return the ResponseEntity with status 200 (ok) and with body filename of the uploaded file
      */
-    @PostMapping("lectures/{lectureId}/attachment-units/split")
+    @PostMapping("lectures/{lectureId}/process-units/upload")
```

Comment:
I do not agree with this change. This directly reverts my change in response to security concerns.
I don't see why it would be necessary to split this in two parts again for this to work. Could you please elaborate on that? If needed, we can also have a meeting to discuss this.

Reply:
A seperate method for the upload is needed because otherwise we need to upload the file everytime the keywords (for slides to removed) are queried. I could keep the split endpoint as-is though. Can you elaborate more on the security concers or maybe send a link to the PR?

Reply:
Resolved after a private conversation; was a misunderstanding about how the feature worked exactly.